ArcGISRuntimeSDKQt_CppSamples/Scenes/GetElevationAtPoint/GetElevationAtPoint.cpp

```diff
@@ -0,0 +1,134 @@
+
+  // Add the marker to the graphics overlay so it will be displayed. Graphics overlay is attached to the sceneView in ::setSceneView()
+  m_graphicsOverlay->graphics()->append(m_elevationMarker);
+}
+
+GetElevationAtPoint::~GetElevationAtPoint() = default;
+
+void GetElevationAtPoint::init()
+{
+  // Register classes for QML
+  qmlRegisterType<SceneQuickView>("Esri.Samples", 1, 0, "SceneView");
+  qmlRegisterType<GetElevationAtPoint>("Esri.Samples", 1, 0, "GetElevationAtPointSample");
+}
+
+SceneQuickView* GetElevationAtPoint::sceneView() const
+{
+  return m_sceneView;
+}
+
+// Set the view (created in QML)
+void GetElevationAtPoint::setSceneView(SceneQuickView* sceneView)
+{
+  if (!sceneView || sceneView == m_sceneView)
+  {
+    return;
+  }
+
+  m_sceneView = sceneView;
+  m_sceneView->setArcGISScene(m_scene);
+
+  // Create a camera, looking at the Himalayan mountain range.
+  const double latitude = 28.4;
+  const double longitude = 83.9;
+  const double altitude = 10000.0;
+  const double heading = 10.0;
+  const double pitch = 80.0;
+  const double roll = 0.0;
+  Camera camera(latitude, longitude, altitude, heading, pitch, roll);
+
+  // Set the sceneview to use above camera, waits for load so scene is immediately displayed in appropriate place.
+  m_sceneView->setViewpointCameraAndWait(camera);
+
+  // Append the graphics overlays to the sceneview, so we can visualise elevation on click
+  m_sceneView->graphicsOverlays()->append(m_graphicsOverlay);
+
+  // Hook up clicks into the 3d scene to below behaviour that displays marker & elevation value.
+  connect(sceneView, &SceneQuickView::mouseClicked, this, &GetElevationAtPoint::displayElevationOnClick);
+
+  emit sceneViewChanged();
+}
+
+void GetElevationAtPoint::displayElevationOnClick(QMouseEvent& mouseEvent)
+{
+  // Convert clicked screen position to position on the map surface.
+  const Point baseSurfacePos = m_sceneView->screenToBaseSurface(mouseEvent.x(), mouseEvent.y());
+
+  // Connect to callback for elevation query, which places marker and sets elevation
+  connect(m_scene->baseSurface(), &Surface::locationToElevationCompleted,
+          this, [baseSurfacePos, this](QUuid /*taskId*/, double elevation)
+  {
+    // Place the elevation marker circle at the clicked position
+    m_elevationMarker->setGeometry(baseSurfacePos);
+    m_elevationMarker->setVisible(true);
+
+    // Assign the elevation value. UI is bound to this value, so it updates to display new elevation.
+    m_elevation = elevation;
+
+    // Notify of property changes
+    emit elevationChanged(elevation);
+    emit elevationQueryRunningChanged();
+  });
+
+  // Invoke get elevation query
+  m_elevationQueryTaskWatcher = m_scene->baseSurface()->locationToElevation(baseSurfacePos);
+  emit elevationQueryRunningChanged();
+}
+
+double GetElevationAtPoint::elevation() const
+{
+  return m_elevation;
+}
+
+bool GetElevationAtPoint::elevationQueryRunning() const
+{
+  return !m_elevationQueryTaskWatcher.isDone();
```

Comment:
I think there is also a cancelled and (possibly invalid) state which I guess this should check?

Reply:
Mmh fair point. Is now 
`return !(m_elevationQueryTaskWatcher.isDone() || m_elevationQueryTaskWatcher.isCanceled());`